src/test/java/saopig/command/ListCommandTest.java

```diff
@@ -0,0 +1,75 @@
+package saopig.command;
+
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import saopig.Storage;
+import saopig.Ui;
+import saopig.task.Task;
+import saopig.task.TaskList;
+
+public class ListCommandTest {
+    private TaskList mockTaskList;
+    private Ui mockUi;
+    private ListCommand listCommand;
+    private Storage mockStorage;
+
+    @BeforeEach
+    void setUp() {
+        mockTaskList = mock(TaskList.class);
+        mockUi = mock(Ui.class);
+        mockStorage = mock(Storage.class);
+    }
+
+    @Test
+    void testListTasksWithEmptyTaskList() {
+        when(mockTaskList.getTasks()).thenReturn(new ArrayList<>());
+        listCommand = new ListCommand("", 0);
+        listCommand.listTasks(mockTaskList, mockUi);
+        verify(mockUi).printMessage(contains("no tasks yet"));
+    }
+
+    @Test
+    void testListTasksWithNonEmptyTaskList() {
+        Task mockTask = mock(Task.class);
+        when(mockTask.toString()).thenReturn("Task Details");
+        when(mockTaskList.getTasks()).thenReturn(new ArrayList<>(Arrays.asList(mockTask, mockTask)));
+        listCommand = new ListCommand("", 0);
+        listCommand.listTasks(mockTaskList, mockUi);
+        verify(mockUi, times(2)).printMessage(contains("Task Details"));
+    }
+
+    @Test
+    void testListTasksOnDateInvalidInputLength() {
```

Comment:
Maybe for some of the tests with longer and cryptic names you could use the alternative naming format unitBeingTested_descriptionOfTestInputs_expectedOutcome? So the reader has a clearer idea of the test and expected outcome.